rmgpy/data/kinetics/family.py

```diff
@@ -1906,4 +1906,64 @@ def getLabeledReactantsAndProducts(self, reactants, products):
             return (None, None)
         else:
             raise Exception('You have {0} reactants, which is unexpected!'.format(len(reactants)))
+        
+    def addAtomLabelsForReaction(self, fam_rxn):
+        """
+        Apply atom labels on a reaction using the appropriate atom labels from this this reaction family.  
+        The reaction's reactants and products must be lists of Molecule objects.
+        The reaction is modified to use Species objects containing the labeled molecules after this function.
+        """
+        # prepare fam_rxts_mol and fam_pdts_mol
+        reactants = [rxt for rxt in fam_rxn.reactants]            
+        products = [pdt for pdt in fam_rxn.products]
+        # prepare fam_forward_template always forward for fam_rxn
+        # because __createReaction always make sure fam_rxn is in forward style
+        family = self
+        template = family.forwardTemplate
+        # Unimolecular reactants: A --> products
+        if len(reactants) == 1 and len(template.reactants) == 1:
+            labeledReactants, labeledProducts = family.getLabeledReactantsAndProducts(reactants, products)
```

Comment:
a large part of the conditionals for uni and bimolecular reactions can be refactored to appear only once in the method, am I correct?